src/main/java/gift/service/AuthService.java

```diff
@@ -0,0 +1,85 @@
+package gift.service;
+
+import gift.exception.UserErrorCode;
+import gift.exception.UserException;
+import gift.model.User;
+import gift.model.dto.UserRequestDto;
+import gift.repository.UserDao;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthService {
+
+    private static final String secretKey = "Yn2kjibddFAWtnPJ2AFlL8WXmohJMCvigQggaEypa5E=";
+    private int accessTokenExpMinutes = 5;
+
+    private final UserDao userDao;
+
+    public AuthService(UserDao userDao) {
+        this.userDao = userDao;
+    }
+
+    public String getToken(UserRequestDto userRequestDto) throws UserException {
+        User user = userDao.selectUserByEmail(userRequestDto.getEmail());
+        if (!matchPassword(userRequestDto.getPassword(), user.getPassword())) {
```

Comment:
user에게 값이 같은지 물어보는 것은 어떨까요?
```suggestion
        if (!user.matchPassword(userRequestDto.getPassword()) {
```

`규칙 9: getter/setter/프로퍼티를 쓰지 않는다.`

Reply:
넵! 객체지향에서의 정보은닉의 중요성이군요!